From 8ccef884226a772d7fbe058350cce85d0298a41f Mon Sep 17 00:00:00 2001
Message-Id: <patch-13.24-8ccef884226-20220309T121511Z-avarab@gmail.com>
In-Reply-To: <cover-00.24-00000000000-20220309T121511Z-avarab@gmail.com>
References: <cover-00.24-00000000000-20220309T121511Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 28 Feb 2022 14:28:50 +0100
Subject: [PATCH 13/24] revisions API users: add release_revisions() to
 wt-status.c edge case
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add a release_revisions() to a use of the "struct rev_info"
has_unstaged_changes() which didn't require the addition of a "goto"
pattern, but needed us to refactor another "return fn(...)" pattern.

This change is split from preceding changes to make similar
conversions in those commits easier to review, as they are all
repetitions of the same pattern.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 wt-status.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/wt-status.c b/wt-status.c
index a68731ea0f7..89289e52603 100644
--- a/wt-status.c
+++ b/wt-status.c
@@ -2537,7 +2537,9 @@ int has_unstaged_changes(struct repository *r, int ignore_submodules)
 	rev_info.diffopt.flags.quick = 1;
 	diff_setup_done(&rev_info.diffopt);
 	result = run_diff_files(&rev_info, 0);
-	return diff_result_code(&rev_info.diffopt, result);
+	result = diff_result_code(&rev_info.diffopt, result);
+	release_revisions(&rev_info);
+	return result;
 }
 
 /**
-- 
2.35.1.1295.g6b025d3e231

